File: comments/src/main/java/net/particify/arsnova/comments/event/RoomDuplicatedEvent.java
package net.particify.arsnova.comments.event;

import java.util.UUID;

class RoomDuplicatedEvent {
  private UUID originalRoomId;
  private UUID duplicatedRoomId;

  public UUID getOriginalRoomId() {
    return originalRoomId;
  }

  public void setOriginalRoomId(final UUID originalRoomId) {
    this.originalRoomId = originalRoomId;
  }

  public UUID getDuplicatedRoomId() {
    return duplicatedRoomId;
  }

  public void setDuplicatedRoomId(final UUID duplicatedRoomId) {
    this.duplicatedRoomId = duplicatedRoomId;
  }
}

File: core4/src/main/kotlin/net/particify/arsnova/core4/room/event/RoomDuplicatedEvent.kt
/* Copyright 2025-2026 Particify GmbH
 * SPDX-License-Identifier: MIT
 */
package net.particify.arsnova.core4.room.event

import java.util.UUID
import org.jmolecules.event.annotation.DomainEvent
import org.jmolecules.event.annotation.Externalized

const val ROOM_DUPLICATED_DESTINATION = "backend.event.room.duplicated"

@DomainEvent
@Externalized(target = ROOM_DUPLICATED_DESTINATION)
data class RoomDuplicatedEvent(val originalRoomId: UUID, val duplicatedRoomId: UUID) : RoomEvent
